pipeline/scripts/13-excelExport.R

```diff
@@ -534,4 +534,43 @@ if (z_score == 1) {
     write.table(pos_contr_warning, file = paste(outdir, "positive_controls_warning.txt", sep = "/"), row.names = FALSE, col.names = FALSE, quote = FALSE)
   }}
 
+### MISSING M/Z CHECK
+# check the outlist_identified_(negative/positive).RData files for missing m/z values and mention in the results mail
+
+# Load the outlist_identified files + remove the loaded files
+load(paste0(outdir,"/outlist_identified_negative.RData"))
+outlist.ident.neg <- outlist.ident
+load(paste0(outdir,"/outlist_identified_positive.RData"))
+outlist.ident.pos <- outlist.ident
+rm(outlist.ident)
+rm(outlist.not.ident)
+
+# check for missing m/z in negative and positive mode
+mode <- c("Negative", "Positive")
+index <- 1
+results_ident <- c() #empty results list
+outlist_ident_list <- list(outlist.ident.neg, outlist.ident.pos)
+for(outlist.ident in outlist_ident_list){
+  current_mode <- mode[index]
+  # retrieve all unique m/z values in whole numbers and check if all are available
+  mz_values <- as.numeric(unique(format(outlist.ident$mzmed.pgrp, digits=0)))
+  mz_range = seq(70, 599, by=1) #change accordingly to the machine m/z range. default = 70-600
```

Comment:
Handig om in achterhoofd te houden voor de Urine grotere massa ranges: een dev branch aanmaken voor het veranderen van deze regel, om de check te laten lopen van 70 - 1050. Is nu nog in ontwikkeling, dus lijkt me (nog) niet nodig om in te bouwen als optie in het script.